coalib/parsing/InvalidFilterException.py

```diff
@@ -1,2 +1,17 @@
 class InvalidFilterException(Exception):
-    pass
+    def __init__(self, filter_name, available_filters):
+        self.filter_name = filter_name
+        self.available_filters = available_filters
+        if self.available_filters is None:
+            message = 'Invalid Filter'
+
+        if self.available_filters is not None:
+            message = {'Invalid filter. Available filters are : ' +
+                       ' ,'.join(self.available_filters)}
+
+        else:
```

Comment:
Here `else` will be executed if `self.available_filters is None` which is the same condition as  stated on line 5.